### ISSUE TITLE
Inconsistent select() and rename() behavior

### ISSUE THREAD

**Issue by jmobrien** [r-lib/tidyselect#90]
Passing a named list to select() performs select/rename behavior, but doing the same with rename() produces the confusing error "All arguments must be named"

It does work when "!!!" is applied, but I'm unclear what's different about the structure here, and the message doesn't really provide insight.

```r
newnames <- c("Sepal.Length", "Petal.Length") %>% 
  setNames(c("Sepal_Length", "Petal_Length"))

iris %>% select(newnames) %>% names
iris %>% rename(newnames) %>% names

iris %>% select(!!!newnames) %>% names
iris %>% rename(!!!newnames) %>% names

```

**Comment by lionel-** [comment]
We're going to deprecate this pattern:

```r
iris %>% select(newnames)
```

The correct way to do it will be one of these (first one is recommended):

```r
iris %>% select(one_of(newnames))
iris %>% select(!!newnames)
```

The reason is that if the data frame contains a `newnames` column, you'll get unwanted results. Conversely if you do `select(x)` because you expect a column named `x`, but there isn't one, and instead a variable `x` is picked up in the environment containing integers or names, you'll get surprising results.

For these reasons, we have decided to make a clear delimitation between expressions referring to columns and expressions referring to the context, like `one_of()`. As usual, unquoting with `!!` is another way of evaluating in the context unambiguously. Note that these considerations only apply to selections, not to actions (as in `mutate()`, `summarise()`, etc). See https://github.com/r-lib/tidyselect/issues/76 for more about this.

The bottom line is that you need to unquote the character vector:

```r
iris %>% rename(!!newnames)
```
